tests/phpunit/src/Commands/DrupalUpdate/DrupalPackageManager.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+use Acquia\Cli\Exception\AcquiaCliException;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class DrupalPackageManager {
+
+  /**
+   * @var OutputInterface
+   */
+  private OutputInterface $output;
+
+  /**
+   * @var array
+   */
+  private array $packageData=[];
+
+  /**
+   * @var bool
+   */
+  private bool $isCoreUpdated;
+
+  /**
+   * @var SymfonyStyle
+   */
+  private SymfonyStyle $io;
+
+  /**
+   * @var array
+   */
+  public array $infoPackageFilesPath = [];
+
+  /**
+   * @var FileSystemUtility
+   */
+  private FileSystemUtility $fileSystemUtility;
+
+  /**
+   * @var DrupalOrgClient
+   */
+  private DrupalOrgClient $drupalOrgClient;
+
+  /**
+   * @var PackageUpdater
+   */
+  private PackageUpdater $packageUpdater;
+
+  /**
+   * @var array
+   */
+  private array $detailPackageData = [];
+
+  /**
+   * @param DrupalOrgClient $drupalOrgClient
+   */
+  public function setDrupalOrgClient(DrupalOrgClient $drupalOrgClient): void {
+    $this->drupalOrgClient = $drupalOrgClient;
+  }
+
+  /**
+   * @param bool $isCoreUpdated
+   */
+  public function setIsCoreUpdated(bool $isCoreUpdated): void {
+    $this->isCoreUpdated = $isCoreUpdated;
+  }
+
+  /**
+   * @param OutputInterface $output
+   */
+  public function setOutput(OutputInterface $output): void {
+    $this->output = $output;
+  }
+
+  /**
+   * @param FileSystemUtility $fileSystemUtility
+   */
+  public function setFileSystemUtility(FileSystemUtility $fileSystemUtility): void {
+    $this->fileSystemUtility = $fileSystemUtility;
+  }
+
+  /**
+   * @param PackageUpdater $package_updater
+   */
+  public function setPackageUpdater(PackageUpdater $package_updater): void {
+    $this->packageUpdater = $package_updater;
+  }
+
+  public function __construct(InputInterface $input, OutputInterface $output) {
+    $this->setIsCoreUpdated(FALSE);
+    $this->setDrupalOrgClient(new DrupalOrgClient($input, $output));
+    $this->setOutput($output);
+    $this->io = new SymfonyStyle($input, $output);
+    $this->setFileSystemUtility(new FileSystemUtility($input, $output));
+    $this->setPackageUpdater(new PackageUpdater($input, $output));
+  }
+
+  /**
+   * @param string $drupal_project_cwd_path
+   * @return mixed
+   */
+  protected function fetchCorePackageVersion(string $drupal_project_cwd_path): mixed {
```

Comment:
Why do files in the tests directory seem to duplicate files in the src directory? For instance this method looks the same as its counterpart in the src directory.